src/test_application/OSPMagnum.cpp

```diff
@@ -160,76 +162,178 @@ osp::active::ActiveScene& OSPMagnum::scene_create(std::string && name)
 void testapp::config_controls(OSPMagnum& rOspApp)
 {
     // Configure Controls
+    //Load toml
+    auto data = toml::parse("settings.toml");
+
+    osp::UserInputHandler& rUserInput = rOspApp.get_input_handler();
+    try {
+        for (const auto& [k, v] : data.as_table()) {
+            std::cout << k << std::endl;
+            std::string primary = toml::find(v, "primary").as_string();
+            std::vector<osp::ButtonVarConfig> controls = parse_control(primary);
+
+            std::string secondary = toml::find(v, "secondary").as_string();
+            std::vector<osp::ButtonVarConfig> secondaryKeys = parse_control(secondary);
+
+            controls.insert(controls.end(), secondaryKeys.begin(), secondaryKeys.end());
+
+            bool holdable = toml::find(v, "holdable").as_boolean();
+
+            //Add user input
+            rUserInput.config_register_control(k, holdable, controls);
+
+        }
+    }
+    catch (std::out_of_range oor) {
```

Comment:
don't catch std::out_of_range.

Catch std::exception.

Also, never catch an exception by value. If you catch an exception by value, you're causing object slicing.

Always catch by const&.